libs/ledger-live-common/src/hw/hooks/useGenuineCheck.ts

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useEffect, useState } from "react";
+import { UserRefusedAllowManager } from "@ledgerhq/errors";
+import type { DeviceId } from "@ledgerhq/types-live";
+import { getGenuineCheckFromDeviceId as defaultGetGenuineCheckFromDeviceId } from "../getGenuineCheckFromDeviceId";
+import type {
+  GetGenuineCheckFromDeviceIdArgs,
+  GetGenuineCheckFromDeviceIdResult,
+  GetGenuineCheckFromDeviceIdOutput,
+} from "../getGenuineCheckFromDeviceId";
+
+export type GenuineState = "unchecked" | "genuine" | "non-genuine";
+export type DevicePermissionState =
+  | "unrequested"
+  | "unlock-needed"
+  | "unlocked"
+  | "requested"
+  | "granted"
+  | "refused";
+
+export type UseGenuineCheckArgs = {
+  isHookEnabled?: boolean;
+  deviceId: DeviceId;
+  lockedDeviceTimeoutMs?: number;
+};
+
+export type UseGenuineCheckDependencies = {
+  getGenuineCheckFromDeviceId?: (
```

Comment:
oh yeah wtf i thought i had a comment for this param, same for `useGetLatestAvailableFirmware`, i need to add a comment for the dependency !

Yes i have a scenario: in LLD, we need this to be defined as a "command" so it can run in the internal thread - see [this PR](https://github.com/LedgerHQ/ledger-live/pull/1026)